src/Compilers/CSharp/Portable/Syntax/LookupPosition.cs

```diff
@@ -175,9 +179,11 @@ internal static bool IsInConstructorParameterScope(int position, ConstructorDecl
                     IsBetweenTokens(position, initializerOpt.ColonToken, nextToken);
             }
 
+            Debug.Assert(constructorDecl.Body != null || (constructorDecl.ExpressionBody != null && constructorDecl.SemicolonToken!= null), 
```

Comment:
Also, SyntaxToken is a struct it cannot be equal to null, perhaps you need to check its kind instead. What it your intent?